src/cow-react/modules/limitOrders/containers/TradeButtons/index.tsx

```diff
@@ -6,28 +6,45 @@ import { tradeFlow, TradeFlowContext } from '@cow/modules/limitOrders/services/t
 import { limitOrdersSettingsAtom } from '@cow/modules/limitOrders/state/limitOrdersSettingsAtom'
 import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimitOrdersTradeState'
 import { useLimitOrdersFormState } from '../../hooks/useLimitOrdersFormState'
-import { limitOrdersTradeButtonsMap, SwapButton } from './limitOrdersTradeButtonsMap'
+import { limitOrdersTradeButtonsMap, SwapButton, WrapUnwrapParams } from './limitOrdersTradeButtonsMap'
 import { limitOrdersConfirmState } from '../LimitOrdersConfirmModal/state'
-import { useToggleWalletModal } from 'state/application/hooks'
+import { useCloseModals, useModalIsOpen, useToggleWalletModal } from 'state/application/hooks'
 import { limitOrdersQuoteAtom } from '@cow/modules/limitOrders/state/limitOrdersQuoteAtom'
 import { useLimitOrdersWarningsAccepted } from '@cow/modules/limitOrders/hooks/useLimitOrdersWarningsAccepted'
 import { PriceImpact } from 'hooks/usePriceImpact'
+import { useWrapCallback } from 'hooks/useWrapCallback'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { transactionConfirmAtom } from '@cow/modules/swap/state/transactionConfirmAtom'
+import { ApplicationModal } from '@src/state/application/reducer'
 
 export interface TradeButtonsProps {
   tradeContext: TradeFlowContext | null
   priceImpact: PriceImpact
+  inputCurrencyAmount: CurrencyAmount<Currency> | null
   openConfirmScreen(): void
 }
 
 export function TradeButtons(props: TradeButtonsProps) {
-  const { tradeContext, openConfirmScreen, priceImpact } = props
+  const { tradeContext, openConfirmScreen, priceImpact, inputCurrencyAmount } = props
   const { expertMode } = useAtomValue(limitOrdersSettingsAtom)
   const formState = useLimitOrdersFormState()
   const tradeState = useLimitOrdersTradeState()
   const setConfirmationState = useSetAtom(limitOrdersConfirmState)
   const toggleWalletModal = useToggleWalletModal()
   const quote = useAtomValue(limitOrdersQuoteAtom)
   const warningsAccepted = useLimitOrdersWarningsAccepted(false)
+  const wrapUnwrapCallback = useWrapCallback(inputCurrencyAmount || undefined)
```

Comment:
This looks a bit weird, is `undefined` really needed here?

Reply:
`useWrapCallback` doesn't allow `null`, fixed it, thanks!